big_scape/comparison/binning.py

```diff
@@ -903,15 +910,22 @@ def legacy_bin_generator(
         if record.product is None:
             continue
 
-        # product hybrids of AS4 and under dealt with here and in legacy_output generate_run_data_js
-        product = ".".join(record.product.split("-"))
+        product = record.product
+
+        if run["hybrids_off"]:
+            record_products = product.split(".")
+
+        else:
+            record_products = [product]
 
-        record_class = legacy_get_class(product)
+        for product in record_products:
+            record_class = legacy_get_class(product)
 
-        try:
-            class_idx[record_class].append(record)
-        except KeyError:
-            class_idx[record_class] = [record]
+            try:
```

Comment:
We should avoid catching an exception when a key does not exist. Instead, use an ```in``` condition:
```
if record_class not in class_idx:
    class_idx[record_class] = []
class_idx[record_class].append(record)
```

you can also use dict.setdefault

```python
# syntax is dict.setdefault(key, value_if_not_exists)
class_idx.setdefault(record_class, [])
class_idx[record_class].append(record)
```